src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -127,30 +121,19 @@
           AfterTargets="CheckTestPlatforms"
           Condition="'$(TestDisabled)'!='true'">
 
-    <Error Condition="'$(RunTestsWithCategories)' != '' And '$(RunTestsWithoutCategories)' != ''"
-           Text="Specifying both RunTestsWithCategories and RunTestsWithoutCategories is not supported please only specify one or the other." />
-
-    <!-- Intersection of TestCategoriesItems and RunTestsWithCategoriesItems -->
-    <ItemGroup Condition="'@(RunTestsWithCategoriesItems)'!=''">
-      <TestCategoriesToRun Include="@(TestCategoriesItems)" Condition="'@(RunTestsWithCategoriesItems)'=='%(Identity)'" />
-    </ItemGroup>
-
-    <!-- Run all test categories except specified -->
-    <ItemGroup Condition="'@(RunTestsWithoutCategoriesItems)'!=''">
-      <TestCategoriesToRun Include="@(TestCategoriesItems)" Exclude="@(RunTestsWithoutCategoriesItems)" />
+    <!-- Default behavior is to disable OuterLoop and failing tests if not specified in WithCategories. -->
+    <ItemGroup>
+      <DefaultNoCategories Include="OuterLoop" />
+      <DefaultNoCategories Include="failing" />
+      <WithoutCategoriesItems Include="@(DefaultNoCategories)" Exclude="@(WithCategoriesItems)" />
+      <WithoutCategoriesItemsDistinct Include="@(WithoutCategoriesItems->Distinct())" />
     </ItemGroup>
 
     <ItemGroup>
-      <RunWithInnerLoopItems Include="@(TestCategoriesToRun)" Condition="'%(TestCategoriesToRun.Identity)'=='InnerLoop'" />
-      <RunWithTraits Condition="'@(RunWithInnerLoopItems)'==''" Include="@(TestCategoriesToRun)" />
-      <RunWithoutTraits Condition="'@(RunWithInnerLoopItems)'!=''" Include="@(TestCategoriesItems)" Exclude="@(TestCategoriesToRun)" />
+      <RunWithTraits Condition="'@(WithCategoriesItems)'!=''" Include="@(WithCategoriesItems)" />
+      <RunWithoutTraits Condition="'@(WithoutCategoriesItemsDistinct)'!=''" Include="@(WithoutCategoriesItemsDistinct)" />
     </ItemGroup>
 
-    <!-- If there is nothing to run, then disable the test -->
-    <PropertyGroup>
-      <TestDisabled Condition="'@(TestCategoriesToRun)'==''">true</TestDisabled>
-    </PropertyGroup>
```

Comment:
Why did your remove this? We shouldn't be trying to run xunit on test projects that don't actually have any tests to run.